geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/index/state/ExtentsEventTimeExpiryFeatureStateFactory.scala

```diff
@@ -0,0 +1,46 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.kafka.index.state
+
+import java.util.concurrent.ScheduledExecutorService
+
+import org.locationtech.geomesa.kafka.index.state.FeatureStateFactory.{ExpiredState, ExtentsExpiryState, FeatureExpiration, FeatureState}
+import org.locationtech.geomesa.utils.cache.Ticker
+import org.locationtech.geomesa.utils.index.SpatialIndex
+import org.opengis.feature.simple.SimpleFeature
+import org.opengis.filter.expression.Expression
+
+/**
+  * Non-point feature index with event time expiration but message time ordering
+  *
+  * @param index spatial index
+  * @param geom geometry attribute
+  * @param executor executor
+  * @param expiry expiry in millis
+  * @param eventTime event time expression
+  */
+class ExtentsEventTimeExpiryFeatureStateFactory(index: SpatialIndex[SimpleFeature],
```

Comment:
I'm tempted to suggest having these Factories in the same file.  Thoughts?

Reply:
i originally had them in one file, but it was kind of confusing to read - i don't really mind either way